app/src/main/java/com/asfoundation/wallet/iab/domain/model/PurchaseData.kt

```diff
@@ -3,24 +3,27 @@ package com.asfoundation.wallet.iab.domain.model
 import android.net.Uri
 import android.os.Parcelable
 import kotlinx.parcelize.Parcelize
+import java.math.BigDecimal
 import kotlin.random.Random
 
 @Parcelize
 data class PurchaseData(
   val referrerUrl: String,
-  val type: String, // this one is strange. we have a TYPE but we use the ESKILLS to build this in OneStepTransactionParser
-  val origin: String?,
-  val skuId: String?,
+  val type: String? = null, // this one is strange. we have a TYPE but we use the ESKILLS to build this in OneStepTransactionParser
+  val origin: String? = null,
+  val skuId: String? = null,
   val domain: String,
-  val productToken: String?,
-  val payload: String?,
-  val callbackUrl: String?,
-  val orderReference: String?,
-  val signature: String?,
-  val value: String?, // originalOneStepValue in previous implementation
-  val currency: String?, // originalOneStepCurrency in previous implementation
-  val oemId: String?,
-  val oemPackage: String?,
+  val productToken: String? = null,
+  val payload: String? = null,
+  val callbackUrl: String? = null,
+  val orderReference: String? = null,
+  val signature: String? = null,
+  val value: String? = null, // originalOneStepValue in previous implementation
```

Comment:
I think we should change this name to whatever it really is, because we have a field named value and another named amount, which creates some confusion during payment or to obtain the correct value of the item